Source/CarthageKit/Xcode.swift

```diff
@@ -1181,6 +1181,67 @@ public func parseSecuritySigningIdentities(securityIdentities securityIdentities
 		.ignoreNil()
 }
 
+public func projectsInDirectory(directoryURL: NSURL, configuration: String, platforms: Set<Platform> = []) -> SignalProducer<(ProjectLocator, [String]), CarthageError> {
+	let directoryLength = directoryURL.pathComponents!.count
+	return locateProjectsInDirectory(directoryURL)
+		.filter { project in
+			let fileURL = project.fileURL
+			let fileComponentsLength = fileURL.pathComponents!.count
+			return fileComponentsLength - directoryLength < 3 // within 2 subdirectories. Because we actually have a test for this.
```

Comment:
It is not needed if we just update (or remove) the Carthage directories of the test fixtures.

However, it would probably be a good idea to cull out the Carthage directory - in case someone accidentally checks in their Carthage directory into source control.

In either case, it should be changed (or removed).